### PR TITLE
[WIP] Add lots of service stubs, fixes and take over mailwl's http:C pull request

### PR DESCRIPTION
**Pull request by ninjfloof**
This pull request adds the following:
1. Friends services nickname by console name.
2. ACT services stubs.
3. Take over [mailwl's http:C](https://github.com/citra-emu/citra/pull/3334) pull request.
And more...

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3454)

### CODE REVIEW

**Comment by ninjfloof** [comment]
What needs to be fixed/updated/corrected:
1. Use default Mii name for friend services nickname, instead of console name.
2. Give unknown ACT functions a name.
3. Add client certificate curlopt into CPR and http:C.
4. Add support for POSTs, PUTs and etc.
5. Move to the new IPC handler.

File: src/core/hle/service/apt/apt.cpp [hunk 1]
```diff
@@ -233,6 +233,14 @@ void Module::Interface::NotifyToWait(Kernel::HLERequestContext& ctx) {
     LOG_WARNING(Service_APT, "(STUBBED) app_id=%u", app_id);
 }
 
+void Module::Interface::ReplySleepQuery(Kernel::HLERequestContext& ctx) {
+    IPC::RequestParser rp(ctx, 0x43, 1, 0); // 0x430040
```
Comment (valentinvanelslande) [inline src/core/hle/service/apt/apt.cpp]: IPC::RequestParser rp(ctx, 0x3E, 2, 0); // 0x003E0080

Comment (ninjfloof) [reply src/core/hle/service/apt/apt.cpp]: Thanks, I forgot about that.

**Comment by mailwl** [comment]
i pushed last changes to http, maybe it helps? https://github.com/citra-emu/citra/pull/3334/commits/f7418dfbf89ee6f4a832709707b3c040afb3361a

**Comment by adityaruplaha** [comment]
Well, @ninjfloof , can you please elaborate on what this does from the users' point of view. I guess this adds a framework for future Internet connectivity implementations?

**Comment by BreadFish64** [comment]
@adityaruplaha if you want to ask about this as a user please contact them by some other method.

**Comment by adityaruplaha** [comment]
@BreadFish64 I would have but I don't have any contact details. @ninjfloof5 are you on Discord?

**Comment by ninjfloof** [comment]
Yes I'm on Discord. Cyuubi (Cyu)#4863

**Comment by ninjfloof** [comment]
Closing this PR, going to redo a lot of it soon. Stay tuned.

**Pull request closed without merging**